### PR TITLE
pkg/prometheus/promcfg*.go: Fix regex in relabel_configs

### LINKED ISSUE #4391: matchExpressions in relabel_configs have a bug (regex)
We must take a values for labels (in regex) in "()" to be compatible with RE2 rules. For example:

we have next selector in Pod/ServiceMonitor
```
selector:                       
matchExpressions:       
- key: app              
  operator: In          
  values:               
  - helm-controller     
  - source-controller   
  - kustomize-controller
```

old prom-operator (0.50.0):
```
  - source_labels: [__meta_kubernetes_pod_label_app]
    separator: ;
    regex: helm-controller|source-controller|kustomize-controller
    replacement: $1
    action: keep
```
new operator (0.52.0)
```
  - source_labels: [__meta_kubernetes_pod_label_app, __meta_kubernetes_pod_labelpresent_app]
    separator: ;
    regex: helm-controller|source-controller|kustomize-controller;true
    replacement: $1
    action: keep
```

This regex:
    **regex: helm-controller|source-controller|kustomize-controller;true**

is invalid. It must be:
    **regex: (helm-controller|source-controller|kustomize-controller);true**

### PR DESCRIPTION
**Pull request by slashpai**
Take values for labels (in regex) in "()" to be compatible with RE2 rules

Fixes #4391

Signed-off-by: Jayapriya Pai <slashpai9@gmail.com>

## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [x] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Fixed regex in relabel_configs
```

### CODE REVIEW

**Comment by slashpai** [comment]
@fpetkovski @simonpasquier Please review

**Comment by fpetkovski** [comment]
Thanks @slashpai , could you just leave a short description in the PR?

**Review by fpetkovski — APPROVED** [review]
@simonpasquier do we want to cut a patch release?

**Comment by sgolod** [comment]
> @simonpasquier do we want to cut a patch release?

that would be great, since monitors with matchExpression do not scraped right now (with 0.52.0 version)

**Comment by slashpai** [comment]
We didn't have a test for matchExpression else we should have found this issue. I will try to add one

**Comment by paulfantom** [comment]
> We didn't have a test for matchExpression else we should have found this issue. I will try to add one

Let's add the test before merging this PR.

**Review by PhilipGough — APPROVED** [review]
/lgtm

File: pkg/prometheus/promcfg_test.go [hunk 1]
```diff
@@ -2036,6 +2036,13 @@ func TestEnforcedNamespaceLabelServiceMonitor(t *testing.T) {
 						MatchLabels: map[string]string{
 							"foo": "bar",
 						},
+						MatchExpressions: []metav1.LabelSelectorRequirement{
```
Comment (slashpai) [inline pkg/prometheus/promcfg_test.go:2039]: Added this to test with both matchLabel and matchExpression together

**Comment by slashpai** [comment]
I am trying to add an e2e test as well

File: pkg/prometheus/promcfg.go [hunk 2]
```diff
@@ -980,13 +980,13 @@ func (cg *ConfigGenerator) generateProbeConfig(
 				relabelings = append(relabelings, yaml.MapSlice{
 					{Key: "action", Value: "keep"},
 					{Key: "source_labels", Value: []string{"__meta_kubernetes_ingress_label_" + sanitizeLabelName(exp.Key), "__meta_kubernetes_ingress_labelpresent_" + sanitizeLabelName(exp.Key)}},
-					{Key: "regex", Value: fmt.Sprintf("%s;true", strings.Join(exp.Values, "|"))},
+					{Key: "regex", Value: fmt.Sprintf("(%s);true", strings.Join(exp.Values, "|"))},
 				})
 			case metav1.LabelSelectorOpNotIn:
 				relabelings = append(relabelings, yaml.MapSlice{
 					{Key: "action", Value: "drop"},
 					{Key: "source_labels", Value: []string{"__meta_kubernetes_ingress_label_" + sanitizeLabelName(exp.Key)}},
```
Comment (simonpasquier) [inline pkg/prometheus/promcfg.go:988]: shouldn't we have `__meta_kubernetes_ingress_label_present_<label>` too?

**Comment by simonpasquier** [comment]
Yes lets have a v0.52.1 release.

File: pkg/prometheus/promcfg.go (hunk 2 above)
Comment (slashpai) [reply pkg/prometheus/promcfg.go]: updated

**Review by simonpasquier — APPROVED** [review]
lgtm, leaving merge to @fpetkovski since he's the 0.52 release shepherd :)

**Comment by slashpai** [comment]
Since I cherry-picked against 0.52 release branch docs test is failing since branch was renamed to main. I guess that will be taken care while releasing

**Comment by simonpasquier** [comment]
> Since I cherry-picked against 0.52 release branch docs test is failing since branch was renamed to main. I guess that will be taken care while releasing

can you cherry-pick 3751ba67a1c1f1f4c98d47a03beb5f72d3db35b5 in this PR? It should allow to merge back release-0.52 into main without conflict.

**Comment by slashpai** [comment]
done

**Comment by fpetkovski** [comment]
Thanks, I'll merge this later today and start with the patch release.

**Pull request merged**